apps/server/src/modules/user-login-migration/service/user-login-migration-rollback.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Account, AccountService } from '@modules/account';
+import { UserService } from '@modules/user';
+import { Injectable } from '@nestjs/common';
+import { UserDO, UserLoginMigrationDO } from '@shared/domain/domainobject';
+import { EntityId } from '@shared/domain/types';
+import { Logger } from '@src/core/logger';
+import {
+	UserLoginMigrationNotFoundLoggableException,
+	UserMigrationRollbackSuccessfulLoggable,
+	UserNotMigratedLoggableException,
+} from '../loggable';
+import { UserLoginMigrationService } from './user-login-migration.service';
+
+@Injectable()
+export class UserLoginMigrationRollbackService {
+	constructor(
+		private readonly userService: UserService,
+		private readonly accountService: AccountService,
+		private readonly userLoginMigrationService: UserLoginMigrationService,
+		private readonly logger: Logger
+	) {}
+
+	public async rollbackUser(targetUserId: EntityId): Promise<void> {
+		const user: UserDO = await this.userService.findById(targetUserId);
+		const account: Account = await this.accountService.findByUserIdOrFail(targetUserId);
+		const userLoginMigration: UserLoginMigrationDO | null = await this.userLoginMigrationService.findMigrationBySchool(
+			user.schoolId
+		);
+
+		if (!userLoginMigration) {
+			throw new UserLoginMigrationNotFoundLoggableException(user.schoolId);
+		}
+
+		if (!user.lastLoginSystemChange) {
```

Comment:
Maybe we should create a function at the user object which wraps the prop. 
Something like isUserMigrated().. to have more "logic" at the domain object

Reply:
I agree, but the UserDO is not one of the new DOs and would have to be refactored in order to work with functions. This should be a separate refactoring, since it can break in a lot of places.